crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -28,13 +28,23 @@ pub fn match_annotated_subscript<'a>(
     expr: &Expr,
     model: &SemanticModel,
     typing_modules: impl Iterator<Item = &'a str>,
+    extend_annotated_subscripts: &[String],
 ) -> Option<SubscriptKind> {
     if !matches!(expr, Expr::Name(_) | Expr::Attribute(_)) {
         return None;
     }
 
     model.resolve_call_path(expr).and_then(|call_path| {
-        if SUBSCRIPTS.contains(&call_path.as_slice()) {
+        let extend_annotated_subscripts: Vec<CallPath> = extend_annotated_subscripts
+            .iter()
+            .map(|target| from_qualified_name(target))
+            .collect();
```

Comment:
Should this be computed in the setting initialization?
I copied `extend_immutable_calls` implementation.